vectorizers/em_token_cooccurrence.py

```diff
@@ -1433,7 +1433,12 @@ def transform(self, X):
     def reduce_dimension(self, dimension=150):
         check_is_fitted(self, ["column_label_dictionary_"])
 
-        self.reduced_matrix_ = normalize(self.cooccurrences_, axis=1, norm="l1")
+        if self.n_iter <= 1:
+            self.reduced_matrix_ = normalize(self.cooccurrences_, axis=1, norm="l1")
+            self.reduced_matrix_ = normalize(self.reduced_matrix_, axis=1, norm="l1")
+        else:
+            self.reduced_matrix_ = normalize(self.cooccurrences_, axis=1, norm="l1")
+
```

Comment:
sorry, I'm missing what this this is doing.

Reply:
If we haven't performed any EM then the columns have not been normalized, so we need to first column normalize then row normalize. It was easiest to just assign to self.reduced_matrix_ with the column normalization and then do the row normalization in the first case, so that means the row normalization occurs to a different matrix in the two cases.

Reply:
Maybe i need more coffee but it looks to me like you are normalizing the rows twice. Don't you need to change to axis=0 for one of these calls?